packages/store-sync/src/zustand/createStorageAdapter.ts

```diff
@@ -139,7 +138,21 @@ export function createStorageAdapter<tables extends Tables>({
             // TODO: warn if no table
 
             // TODO: update decodeKey to use more recent types
-            const key = decodeKey(getSchemaTypes(getKeySchema(table)) as KeySchema, rawRecord.keyTuple);
+            const keySchema = getSchemaTypes(getKeySchema(table));
+            const keyTupleLength = rawRecord.keyTuple.length;
+            const keySchemaLength = Object.keys(keySchema).length;
+            if (keySchemaLength !== keyTupleLength) {
+              debug(
+                `key tuple length ${keyTupleLength} does not match key schema length ${keySchemaLength}, skipping record`,
+                {
+                  table,
+                  rawRecord,
+                },
+              );
+              return undefined;
```

Comment:
is returning undefined required or can we just return?

```suggestion
              return;
```